packages/block-editor/src/components/iframe/use-scale-canvas.js

```diff
@@ -29,17 +38,17 @@ function calculateScale( {
 /**
  * Compute the next scrollTop position after scaling the iframe content.
  *
- * @param {number} scrollHeight   Scaled height of the current iframe content.
- * @param {Object} transitionFrom The starting point of the transition
- * @param {Object} transitionTo   The ending state of the transition
- * @return {number} The next scrollTop position after scaling the iframe content.
+ * @param {TransitionState} transitionFrom Starting point of the transition
+ * @param {TransitionState} transitionTo   Ending state of the transition
+ * @return {number} Next scrollTop position after scaling the iframe content.
  */
-function computeScrollTopNext( scrollHeight, transitionFrom, transitionTo ) {
```

Comment:
While it is another param to pass in, I don't think we should include `scrollHeight` in the `TransitionState` object. The `TransitionState` doesn't need to know anything about the `scrollHeight`. We only ned the `scrollHeight` to compute the `maxScrollTop`, which seems like a separate responsibility.

Reply:
I think making the function definition less confusing with just the `from` and `to` positional arguments is a worthwhile reason to update.

The other `iframeDocument.documentElement` properties like `clientHeight` are also added to the `transitionFrom.current` `TransitionState`.

And it could still make sense in the transition state because `scrollTop` is part of the transition state and `maxScrollTop` is part of `scrollTop`.